x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/persistence/ProfilingIndexManager.java

```diff
@@ -249,35 +254,35 @@ private void deleteIndices(final String[] indices, final ActionListener<Acknowle
         executeAsync("delete", request, listener, (req, l) -> client.admin().indices().delete(req, l));
     }
 
-    enum OnVersionBump {
+    public enum OnVersionBump {
         DELETE_OLD,
         KEEP_OLD
     }
 
     /**
      * An index that is used by Universal Profiling.
      */
-    static class ProfilingIndex implements ProfilingIndexAbstraction {
+    public static class ProfilingIndex implements ProfilingIndexAbstraction {
         private final String namePrefix;
         private final int version;
         private final String generation;
         private final OnVersionBump onVersionBump;
         private final List<Migration> migrations;
 
-        public static ProfilingIndex regular(String name, int version, OnVersionBump onVersionBump) {
+        static ProfilingIndex regular(String name, int version, OnVersionBump onVersionBump) {
```

Comment:
While we needed to declare the class itself public, there is no need for code outside the `persistence` package to create new indices and hence we kept the factory methods package-private.